src/restruct_poc/solver/assemble_constraint_force.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <Kokkos_Core.hpp>
+
+#include "constraints.hpp"
+
+#include "src/restruct_poc/math/quaternion_operations.hpp"
+#include "src/restruct_poc/math/vector_operations.hpp"
+#include "src/restruct_poc/types.hpp"
+
+namespace openturbine {
+
+struct AssembleConstraintForce {
```

Comment:
I'm thinking we should go ahead and split this function in a similar way to how I did calculate_constraint_residual_gradient (i.e. turn the contents of the if condition into its own kernel).  I know it might seem silly with just one constraint type, but it will set us up for adding more types in the future and makes testing nice.